src/include/duckdb/catalog/catalog_entry/table_catalog_entry.hpp

```diff
@@ -114,10 +115,18 @@ class TableCatalogEntry : public StandardEntry {
 	//! Returns true, if the table has a primary key, else false.
 	bool HasPrimaryKey() const;
 
+	//! Returns the rowid type of this table
+	LogicalType GetRowIdType() {
```

Comment:
Can we make this a virtual method instead that returns `LogicalType::ROW_TYPE`? Then we don't need to keep this as a member or take this in the constructor.